src/pages/StocktakesPage.js

```diff
@@ -54,95 +42,111 @@ export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch
     isAscending: false,
     modalKey: '',
     hasSelection: false,
-    currentUser,
-    reduxDispatch,
     usingPrograms: getAllPrograms(Settings, UIDatabase).length > 0,
-  });
+  };
+};
+
+export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch, navigation }) => {
+  const [state, dispatch, instantDebouncedDispatch] = usePageReducer(
+    routeName,
+    {},
+    stateInitialiser
+  );
 
   const {
     data,
     dataState,
     sortBy,
     isAscending,
-    columns,
+    searchTerm,
     modalKey,
     hasSelection,
     usingPrograms,
+    keyExtractor,
+    columns,
+    PageActions,
   } = state;
 
-  const { PROGRAM_STOCKTAKE } = MODAL_KEYS;
+  const callback = useCallback(() => dispatch(PageActions.refreshData()), []);
+  // Listen to sync changing stocktake data - refresh if there are any.
+  useSyncListener(callback, ['Stocktake']);
+  // Listen to navigation focusing this page - fresh if so.
+  useNavigationFocus(callback, navigation);
+
+  const onRowPress = useCallback(stocktake => reduxDispatch(gotoStocktakeEditPage(stocktake)), []);
+  const onFilterData = value => dispatch(PageActions.filterData(value));
+  const onCancelDelete = () => dispatch(PageActions.deselectAll());
+  const onConfirmDelete = () => dispatch(PageActions.deleteStocktakes());
+  const onCloseModal = () => dispatch(PageActions.closeModal());
+
+  const onNewStocktake = () => {
+    if (usingPrograms) return dispatch(PageActions.openModal(MODAL_KEYS.PROGRAM_STOCKTAKE));
+    return reduxDispatch(gotoStocktakeManagePage({ stocktakeName: '' }));
+  };
 
-  const getAction = useCallback((colKey, propName) => {
+  const getAction = (colKey, propName) => {
     switch (colKey) {
       case 'remove':
-        if (propName === 'onCheckAction') return selectRow;
-        return deselectRow;
+        if (propName === 'onCheckAction') return PageActions.selectRow;
```

Comment:
is having a `COLUMN_KEYS` constant overkill, to remove the `case 'remove':` in place of `case COLUMN_KEYS.REMOVE:`

maybe `return propName === 'onCheckAction' ? PageActions.selectRow : PageActions.deselectRow` or maybe thats ugly

Reply:
probably is overkill. but I like the idea all the same

Reply:
In general, I think all string literals should have constants. It reduces errors and indicates clearly the "interface" that the field implements. I like this because it also can be used for run-time type checking:

```
if (Object.values(COLUMN_KEYS).includes(key)) {
  // Do stuff.
} else {
  // Handle error.
}
```